robot-core/src/main/java/org/obolibrary/robot/IOHelper.java

```diff
@@ -273,6 +283,34 @@ protected static boolean cleanTDB(String tdbDir) {
     return success;
   }
 
+  /**
+   * Get a set of parsers for a given ontology format.
+   *
+   * @param format ontology file format as string
+   * @return set of OWLParserFactories
+   * @throws IllegalArgumentException on bad format
+   */
+  public static Set<OWLParserFactory> getParsersByFormat(String format) {
+    switch (format) {
+      case "omn":
+        return Sets.newHashSet(new ManchesterOWLSyntaxOntologyParserFactory());
+      case "obo":
+        return Sets.newHashSet(new OBOFormatOWLAPIParserFactory());
+      case "ofn":
+        return Sets.newHashSet(new OWLFunctionalSyntaxOWLParserFactory());
+      case "owx":
+        return Sets.newHashSet(new OWLXMLParserFactory());
+      case "owl":
+        return Sets.newHashSet(new RDFXMLParserFactory(), new RioRDFXMLParserFactory());
```

Comment:
Why do we need to add the RIO parser here? Shouldnt we be more strict?

Reply:
I don't have strong feelings one way or another. I figured it would be used if RDFXML failed without `--input-format` anyway? @jamesaoverton do you have a preference for excluding Rio parsers?